SofaKernel/framework/sofa/helper/Quater.inl

```diff
@@ -547,21 +547,24 @@ defaulttype::Vec<3,Real> Quater<Real>::quatToRotationVector() const
 }
 
 
+/// Compute the Euler angles:
+/// Roll: rotation about the X-axis
+/// Pitch: rotation about the Y-axis
+/// Yaw: rotation about the Z-axis
 template<class Real>
 defaulttype::Vec<3,Real> Quater<Real>::toEulerVector() const
 {
-///    Compute the Euler angles:
-///    Roll: rotation about the X-axis
-///    Pitch: rotation about the Y-axis
-///    Yaw: rotation about the Z-axis
-
     Quater<Real> q = *this;
-        q.normalize();
-        defaulttype::Vec<3,Real> vEuler;
-        vEuler[0] = atan2(2*(q[3]*q[0] + q[1]*q[2]) , (1-2*(q[0]*q[0] + q[1]*q[1])));   //roll
-        vEuler[1] = asin(2*(q[3]*q[1] - q[2]*q[0]));                                    //pitch
-        vEuler[2] = atan2(2*(q[3]*q[2] + q[0]*q[1]) , (1-2*(q[1]*q[1] + q[2]*q[2])));   //yaw
-        return vEuler;
+    q.normalize();
+
+    // Cancel numerical drifting by clamping on [-1 ; 1]
+    Real y = std::max(Real(-1.0), std::min(Real(1.0), Real(2.)*(q[3]*q[1] - q[2]*q[0])));
+
+    defaulttype::Vec<3,Real> vEuler;
+    vEuler[0] = atan2(2*(q[3]*q[0] + q[1]*q[2]) , (1-2*(q[0]*q[0] + q[1]*q[1])));   //roll
+    vEuler[1] = asin(y); // pitch
+    vEuler[2] = atan2(2*(q[3]*q[2] + q[0]*q[1]) , (1-2*(q[1]*q[1] + q[2]*q[2])));   //yaw
```

Comment:
as it is done for asin, shouldn't we add a test on atan2?
I'm not export in this, are all values possible for atan2? quick search gave me that : 

> Systems supporting symbolic mathematics normally return an undefined value for atan2(0, 0)

Reply:
ok for me